src/migration.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2015-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+/// Database migration.
+
+use std::path::Path;
+use crate::{options::Options, db::Db, Error, Result, column::ColId};
+
+const COMMIT_SIZE: usize = 10240;
+
+pub fn migrate(from: &Path, mut to: Options) -> Result<()> {
+	let mut path: std::path::PathBuf = from.into();
+	path.push("metadata");
+	let (source_cols, salt) = Options::load_metadata(&path)?;
+	let source_cols = source_cols.ok_or_else(|| Error::Migration("Error loading source metadata".into()))?;
+	if source_cols.len() != to.columns.len() {
+		return Err(Error::Migration("Source and dest columns mismatch".into()));
+	}
+
+	let mut source_options = Options::with_columns(from, source_cols.len() as u8);
+	source_options.columns = source_cols;
+
+	// Make sure we are using the same salt value.
+	to.salt = salt;
+
+	let source = Db::open(&source_options)?;
+	let dest = Db::open(&to)?;
+
+	let mut ncommits: u64 = 0;
+	let mut commit = Vec::with_capacity(COMMIT_SIZE);
+	let mut last_time = std::time::Instant::now();
+	for c in 0 .. source_options.columns.len() as ColId {
+		log::info!("Migrating col {}", c);
+		source.iter_column_while(c, |index, key, rc, mut value| {
+			//TODO: more efficient ref migration
```

Comment:
I went for a more general approach, so that the values could be migrated between value tables, which requires index update. This seems like the simplest way to do it for now.